src/drivers/protocol_splitter/protocol_splitter.cpp

```diff
@@ -147,16 +147,27 @@ class DevCommon : public cdev::CDev
 
 	/*
 	struct Sp2Header {
-		char magic[3];
-		uint8_t type;
-		uint16_t payload_len;
-		uint16_t reserved (align)
+		char magic;                // 'S'
```

Comment:
Any chance we can use this struct throughout the code?

Reply:
I can do that. It may increase the instruction count a bit, but I guess it does not matter so much. It may not make the code more readable in some places, but probably not confusing too much either. I guess the compiler also handles the packet struct assignment into byte stream properly in case of odd pointer address, at least if all struct items are bytes. I'll add a new commit for this, let see how it looks..